frame/handler.ts

```diff
@@ -15,22 +17,22 @@ import {
   RunAdAuctionResponse,
 } from "../lib/shared/protocol";
 import { isArray } from "../lib/shared/types";
+import { runAdAuction } from "./auction";
+import { setInterestGroupAds, deleteInterestGroup } from "./database";
 import { errorMessage } from "./error";
-import { setInterestGroupAds, deleteInterestGroup, getAllAds } from "./storage";
 
 /**
- * TODO
+ * Handles a `MessageEvent` representing a request to the FLEDGE API, and sends
+ * a response via the provided ports if needed.
  *
- * @param callbackForTesting TODO
+ * If an error occurs, it is sent as a raw string to each provided port, in
+ * order to allow the error message to be propagated back to the caller.
```

Comment:
I'm nervous that this could leak details of interest groups?

Reply:
Quite right. Fixed.